src/edu/washington/escience/myria/storage/TupleBatch.java

```diff
@@ -101,8 +101,11 @@ public TupleBatch(final Schema schema, final List<? extends Column<?>> columns,
     this.columns = ImmutableList.copyOf(Objects.requireNonNull(columns, "columns"));
     Preconditions.checkArgument(columns.size() == schema.numColumns(),
         "Number of columns in data must equal the number of fields in schema");
-    for (Column<?> column : columns) {
-      Preconditions.checkArgument(numTuples == column.size(), "Column %s != %s tuples", column.size(), numTuples);
+    for (int i = 0; i < columns.size(); i++) {
+      Column<?> column = columns.get(i);
+      Preconditions.checkArgument(numTuples == column.size(),
+              "Incorrect size for column %s. Expected %s tuples, but found %s tuples.",
+              i, numTuples, column.size());
```

Comment:
I is the column index, not the row. What you have here is correct, the test failed not because of this change. I refreshed it and it passed. 
So I'll revert your following commit. I think "i" is definitely an appropriate name for column index variable :)